src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.GateThread.cs

```diff
@@ -250,7 +250,7 @@ private static void CreateGateThread(PortableThreadPool threadPoolInstance)
                 {
                     if (!created)
                     {
-                        Interlocked.Exchange(ref threadPoolInstance._separated.gateThreadRunningState, 0);
+                        Environment.FailFast("Failed to create Gate Thread");
```

Comment:
Will there have been an exception thrown? If so, might there be a useful message to include in the Fail fast here (eg out of memory)

Reply:
Sorry Dan not sure if I got that, you mean updating the message to "Out Of Memory"? Just in case, I got this output when testing the `Environment.FailFast`.
```
Out of memory.
Process terminated. Failed to create Gate Thread
   at System.Environment.FailFast(System.String)
   at System.Threading.PortableThreadPool+GateThread.CreateGateThread(System.Threading.PortableThreadPool)
   at System.Threading.PortableThreadPool+GateThread.EnsureRunningSlow(System.Threading.PortableThreadPool)
   at System.Threading.PortableThreadPool+GateThread.EnsureRunning(System.Threading.PortableThreadPool)
   at System.Threading.PortableThreadPool.RequestWorker()
   at System.Threading.ThreadPoolWorkQueue.EnsureThreadRequested()
   at System.Threading.ThreadPoolWorkQueue.Enqueue(System.Object, Boolean)
   at System.Threading.ThreadPool.QueueUserWorkItem(System.Threading.WaitCallback, System.Object)
   at System.Threading.ThreadPool.QueueUserWorkItem(System.Threading.WaitCallback)
   at Example.Main()
```

Reply:
Could perhaps change the finally to catch all exceptions and pass the exception into `FailFast`.